packages/zero-vite-plugin/src/index.ts

```diff
@@ -45,19 +48,25 @@ function isZeroRuntimeProcessableFile(fileName: string, transformLibraries: stri
 
 export function zeroVitePlugin(options: ZeroVitePluginOptions) {
   const {
-    cssVariablesPrefix = 'mui',
-    injectDefaultThemeInRoot = true,
     theme,
     babelOptions = {},
     preprocessor = basePreprocessor,
     transformLibraries = [],
     ...rest
   } = options ?? {};
-  const isExtendTheme = !!(theme && typeof theme === 'object' && 'vars' in theme && theme.vars);
-  const varPrefix: string =
-    isExtendTheme && 'cssVarPrefix' in theme
-      ? (theme.cssVarPrefix as string) ?? cssVariablesPrefix
-      : cssVariablesPrefix;
+
+  // create stylesheet as object
+  const stylesheetObj: Interpolation<unknown> = {
+    ':root': theme.generateCssVars().css,
+  };
+  Object.entries(theme.colorSchemes).forEach(([key]) => {
+    stylesheetObj[
+      `${key === 'light' ? ':root, ' : ''}[data-${theme.cssVarPrefix}-color-scheme="${key}"]`
+    ] = theme.generateCssVars(key).css;
+  });
```

Comment:
This whole thing can be a util in `zero-runtime/utils` package as these are duplicated across vite and unplugin and also in future, will need change in only one place regardless of the different plugins.